include/circt/Dialect/FIRRTL/FIRRTLStatements.td

```diff
@@ -375,4 +376,31 @@ def VerifAssertIntrinsicOp : VerifIntrinsicOp<"assert">;
 def VerifAssumeIntrinsicOp : VerifIntrinsicOp<"assume">;
 def VerifCoverIntrinsicOp : VerifIntrinsicOp<"cover">;
 
+//===----------------------------------------------------------------------===//
+// Group Op
+//===----------------------------------------------------------------------===//
+
+def GroupOp : FIRRTLOp<
+  "group",
+  [SingleBlock, NoTerminator, NoRegionArguments,
+   HasParent<"firrtl::FModuleOp, firrtl::GroupOp">,
```

Comment:
`InstanceOp`s can go in groups without restrictions. If that instance is a bound instance or has any child instances that are bound, then this requires special handling during group lowering. (Tracking verifier issue here: https://github.com/llvm/circt/issues/5530)

It's intentional that groups can't be put under `WhenOp` or `MatchOp`. This could probably be made to work so long as groups can be merged. It should also be possible to convert a group under a when to a when under a group. The only thing is that any drivers of sinks outside the group would then be dropped.

I'm going to initially disallow this and then see about allowing it later. The main use case for allowing it is to enable use of a Chisel assert API (which would automatically add a group) involving a declaration under a `WhenOp`.